app/helpers/dashboard_helper.rb

```diff
@@ -21,7 +21,7 @@ def last_next_run(widget)
   end
 
   def column_widgets(dashboard)
-    [:col1, :col2].map { |column| dashboard.key?(column) ? dashboard[column].uniq : [] }
+    [:col1, :col2].map { |column| dashboard&.key?(column) ? dashboard[column].uniq : [] }
```

Comment:
Why would there be a nil dashboard?  Is it possible to add a spec for this?

Reply:
Reason I'm asking is there might need to be other changes such as in the column_data method?

Reply:
In fact, I wonder if this should be refactored to

```suggestion
    [:col1, :col2].map { |column| column_data(dashboard, column) }
```

Reply:
the `column_data` method is not being used anywhere, hence removed this method.

Reply:
added a spec for `column_widgets` method for `dashboard_helper`

Reply:
> Reason I'm asking is there might need to be other changes such as in the column_data method?
added a spec for `column_widgets` method for `dashboard_helper`